reference/events.rst

```diff
@@ -158,6 +158,25 @@ Listener Class Name
 :class:`Symfony\\Component\\HttpKernel\\EventListener\\StreamedResponseListener`     -1024
 ===================================================================================  ========
 
+``kernel.finish_request``
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+**Event Class**: :class:`Symfony\\Component\\HttpKernel\\Event\\FinishRequestEvent`
+
+The purpose of this event is to allow you to reset the global and environmental state of
+the application, when it was changed during the request.
```

Comment:
I would change "when it was changed during the request" to "after a sub-request has finished."

Maybe add the example of the translator listener: Resetting the locale to the locale of the parent request when finishing a sub-request.